### ISSUE TITLE
Gear Compare forgets settings

### ISSUE THREAD

**Issue by oBusk** [seriallos/raidbots-issues#407]
# If your bug is related to an incorrect tooltip, check the Full HTML Results before reporting! https://github.com/seriallos/raidbots-issues/issues/288

**Describe the bug**
If you setup a gear comparison in the gear compare tool, and then uses `ctrl`-click to create the comparison in a new tab, all of your configured sets will be reset, meaning you have to set it all back up to run the same comparison with another Fight style for example.
I'd argue this is a bug since this is a different behaviour from the top gear tool where you can ctrl-click to create the comparison, and maintain the settings so that you can do another comparison without having to set it all back up.

Furthermore, if you hit the "Run sim again" button on the result page, you end up on a gear compare page _without_ any sets configured. If you paste the simc base into the textbox, the sets will re-appear. This is most definitely a bug since the sets are _there_ just not showing until you paste again.

**To Reproduce**

1. Go to "Gear Compare"
2. Paste in any simc addon output
3. Setup one ore more sets to compare.
4. Hold down ctrl/cmd and click on "COMPARE ITEMS"
5. The simulation will start running in a new tab, but the sets you had previously is now gone.
6. Once the simulation is done, go to the resul page
7. Click "Run sim again"
8. You should now be seeing a gear compare page, without any character data or sets
9. Paste the same simc data in again,
10. You should now see the sets that you setup previously re-appear.

**Expected behavior**

- If you start the simulation while holding ctrl, no setting should reset, so that I can tweak something and re-run it without having to set it all back up
- If I click "run sim again", all config should be remembered, and should not have to paste new simc addon data.

**Link a sim with the problem (if available)**

For the secondary "Run sim again" issue, I assume any gear compare will have this issue, but here is one for example https://www.raidbots.com/simbot/report/9jL9yytBpNp4Q3fUp8XHe8

**Character or /simc input**

<details>
<summary>/simc input</summary>

```txt
# Nulldozzer - Arms - 2023-05-24 08:59 - EU/Ragnaros
# SimC Addon 10.1.0-04
# WoW 10.1.0.49679, TOC 100100
# Requires SimulationCraft 1000-01 or newer

warrior="Nulldozzer"
level=70
race=troll
region=eu
server=ragnaros
role=attack
professions=blacksmithing=100/alchemy=100
spec=arms

talents=BcEAoonA2h02U1PVo4204nicFAQplIRkQkkSrkEAAAAQQkIKEQREQiQSoJRIAhECAAAAAAAAJBJJgQ0QA


head=,id=200426,gem_id=192948,bonus_id=6652/9130/7977/7935/8828/1495
neck=,id=193001,gem_id=192948/192985/192948,bonus_id=8836/8840/8902/8960/8784/8782/9405/8790/8846/9365,crafted_stats=49/40,crafting_quality=5
shoulder=,id=202441,bonus_id=6652/9227/7979/9325/1485/8767
back=,id=202573,enchant_id=6604,bonus_id=7979/6652/9222/9220/9324/1481/8767
chest=,id=202446,enchant_id=6625,bonus_id=6652/9231/7979/9329/1488/8767
shirt=,id=18231
tabard=,id=119138
wrist=,id=190502,bonus_id=8836/8840/8902/9405/9373/8793/9379/8960/9366/9414,crafted_stats=49/40,crafting_quality=5
hands=,id=202444,bonus_id=6652/9230/9409/9333/1488/8767
waist=,id=202577,enchant_id=6904,bonus_id=9409/9330/6652/9414/9226/9219/1485/8767
legs=,id=200427,enchant_id=6496,bonus_id=6652/8827/7977/8973/1498/8767
feet=,id=193771,enchant_id=6607,bonus_id=6652/9223/9219/9144/9325/1650/8767
finger1=,id=192999,enchant_id=6556,gem_id=192948,bonus_id=8836/8840/8902/8780/9405/8790/8846/8174/9365,crafted_stats=49/32,crafting_quality=5
finger2=,id=203460,enchant_id=6556,gem_id=204029/204027/204011,gem_bonus_id=9369:1472/9369:1472/9369:1472,bonus_id=1491
trinket1=,id=193701,bonus_id=7977/6652/9144/8973/1637/8767
trinket2=,id=202613,bonus_id=9409/9330/6652/1485/8767
main_hand=,id=202559,enchant_id=6655,bonus_id=9409/6652/9334/1498/8767

### Gear from Bags
#
# Thraexhelm of the Onyx Crucible (434)
# head=,id=202443,bonus_id=42/9414/9229/9409/9332/1491/8767
#
# Master's Wildercloth Fishing Cap (372)
# head=,id=193543,bonus_id=8843/8842/8807,crafting_quality=5
#
# Peaks of the Walking Mountain (415)
# shoulder=,id=200428,bonus_id=6652/7981/8095/8826/1489
#
# Shoulderplates of Planar Isolation (424)
# shoulder=,id=202595,bonus_id=7979/6652/9222/9218/9324/1481/8767
#
# Deeprock Cape (424)
# back=,id=204915,enchant_id=6604,bonus_id=6652/9320/1646/8767
#
# Sarkareth's Abyssal Embrace (424)
# chest=,id=202599,bonus_id=6652/9222/9221/9324/7979/1472/8767
#
# Rockbound Chestguard (428)
# chest=,id=134135,bonus_id=9330/6652/9223/9221/9144/3307/8767
#
# Husk of the Walking Mountain (421)
# chest=,id=200423,enchant_id=6625,bonus_id=7981/6652/8830/1498/8767
#
# Suffused Armplates (405)
# wrist=,id=204761,bonus_id=9314/6652/9414/9223/9220/1527/8767
#
# Allied Wristguard of Companionship (421)
# wrist=,id=190526,enchant_id=6586,gem_id=192948,bonus_id=8836/8840/8902/8960/9405/8846/9365/9436,crafting_quality=5
#
# Gauntlets of the Walking Mountain (415)
# hands=,id=200425,bonus_id=6652/7977/8973/8829/1492
#
# Starless Legguards (424)
# legs=,id=204949,enchant_id=6496,bonus_id=6652/9320/1646/8767
#
# Sharkwater Waders (424)
# legs=,id=158361,enchant_id=6496,bonus_id=9324/6652/9223/9221/9144/3294/8767
#
# Deeprock Ring (402)
# finger1=,id=204917,bonus_id=40/9413/9313/1624/8767
#
# Suffused Band (408)
# finger1=,id=204735,bonus_id=9315/6652/9413/1530/8767
#
# Ring-Bound Hourglass (421)
# finger1=,id=193000,enchant_id=6556,gem_id=192948,bonus_id=8836/8840/8902/8780/9405/8790/8846/9365,crafted_stats=36/40,crafting_quality=5
#
# Onyx Annulet (424)
# finger1=,id=203460,enchant_id=6556,gem_id=204029/204027,gem_bonus_id=9369:1472/9369:1472,bonus_id=1491
#
# Enduring Dreadplate (402)
# trinket1=,id=202616,bonus_id=9411/9313/41/1459/8767
#
# Manic Grieftorch (411)
# trinket1=,id=194308,bonus_id=7980/6652/1485/8767
#
# Storm-Eater's Boon (395)
# trinket1=,id=194302,bonus_id=6652/7979/1472/8767
#
# Decoration of Flame (415)
# trinket1=,id=194299,bonus_id=7981/6652/1498/8767
#
# Whispering Incarnate Icon (408)
# trinket1=,id=194301,bonus_id=6652/7980/1485/8767
#
# Rumbling Ruby (415)
# trinket1=,id=194303,bonus_id=6652/7981/1498/8767
#
# Corruption Borne Headlopper (431)
# main_hand=,id=159654,enchant_id=6643,bonus_id=6652/9147/9327/3301/8767
#
# Obsidian Seared Claymore (421)
# main_hand=,id=190514,enchant_id=6643,bonus_id=8836/8840/8902/9405/8846/8790/9237/9365,crafted_stats=40/36,crafting_quality=5
#
# Maul of the Earthshaper (415)
# main_hand=,id=195489,enchant_id=6655,bonus_id=6652/7981/1498/8767
#
# Scholar's Thinking Cudgel (415)
# main_hand=,id=202568,bonus_id=9321/7979/6652/1472/8767
#
# Heartseeker of Raging Tempests (359)
# main_hand=,id=199307,bonus_id=6652/9340/1481/8766
#
# Neltharic Greatsword (415)
# main_hand=,id=204967,bonus_id=6652/9321/1637/8767
#
# Horn of Raging Tempests (359)
# off_hand=,id=199311,bonus_id=6652/9340/1481/8766
#
# Experiment 1, Kitewing (424)
# off_hand=,id=202571,bonus_id=41/9411/9320/1478/8767

### Weekly Reward Choices
#
# Legplates of Profane Sacrifice
# legs=,id=159443,bonus_id=9380/6652/9223/9221/9144/3311/8767
#
# Magmorax's Fourth Collar
# neck=,id=204397,bonus_id=6652/9414/8783/9409/9332/1485/8767
#
# Legplates of the Onyx Crucible
# legs=,id=202442,bonus_id=6652/9228/9409/9331/1488/8767
#
# Screaming Black Dragonscale
# trinket1=,id=202612,bonus_id=6652/9409/9332/1485/8767

### Additional Character Info
#
# upgrade_currencies=c:1792:50/c:2245:108/i:204195:22/i:204440:1/i:204193:34/i:198059:1/i:190453:2/i:198058:5/i:197921:4/i:198056:1/i:198048:3/i:204196:1
#
# slot_high_watermarks=0:434:434/1:421:421/2:428:428/3:437:437/4:428:428/5:424:424/6:428:428/7:437:437/8:437:437/9:424:424/10:415:415/11:424:424/12:441:441/13:0:0/14:415:415/15:415:415/16:424:424

# Checksum: 7fea1529
```

</details>

**Screenshots**

![image](https://github.com/seriallos/raidbots-issues/assets/13413409/1fec4990-9de5-4db6-8f42-52bb7a5517b1)
After ctrl-clicking, it's shown as loading
![image](https://github.com/seriallos/raidbots-issues/assets/13413409/a265abc1-dab5-4f59-a83a-7f329edebc7a)
And then all the sets are gone and would need to be re-setup
![image](https://github.com/seriallos/raidbots-issues/assets/13413409/909d4cb6-286e-4fa6-a3fb-ae61701d4f5f)

**Comment by seriallos** [comment]
Yeah, Gear Compare is a much older tool and doesn't fully support the ctrl/middle-click functionality due to some old and nasty technical debt. It's on my list to look into at some point but it's a fairly low priority at the moment.
